src/Promise.php

```diff
@@ -166,6 +170,9 @@ private function reject(\Throwable $reason): void
         $this->settle(reject($reason));
     }
 
+    /**
+     * @param PromiseInterface<T>|PromiseInterface<null> $result
```

Comment:
The `PromiseInterface<null>` seems weird here. It looks like this should be `PromiseInterface<never>` for rejected promises? In either case, a short comment would be useful here :+1: